src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -310,8 +313,35 @@ export class ReadingsGraphComponent implements OnDestroy {
     this.getAssetReadings(this.payload);
   }
 
+  updateXAxisTickFormat(length) {
+    // below 1 minute
+    if (length < 60) {
+      this.layout.xaxis.tickformat = '%H:%M:%S.%L';
+    }
+    // 1 minute to 6 hours
+    if (60 <= length && length <= 21600) {
+      this.layout.xaxis.tickformat = '%H:%M:%S';
+    }
+    // 6 hours to 1 day
+    if (21600 < length && length < 86400) {
+      this.layout.xaxis.tickformat = '%H:%M';
+    }
+    // 1 day to 1 week
+    if (86400 <= length && length < 604800) {
+      this.layout.xaxis.tickformat = '%e %b %H:%M';
+    }
+    // 1 week to 6 months
+    if (604800 <= length && length < 15552000) {
+      this.layout.xaxis.tickformat = '%e %b';
+    }
+    // above 6 months
+    if (15552000 <= length) {
+      this.layout.xaxis.tickformat = '%b %y';
+    }
+  }
+
   dragGraph(event: any) {
-    if (event['xaxis.range[0]'] === undefined) {
+    if (event['xaxis.range[0]'] === undefined  || this.numReadings[0] === undefined) {
```

Comment:
can this be `this.numReadings` length check?

Reply:
yes, it can be.